File: data structures/stack/stack.java
public class stack{
    int [] arr = new int[100];
    int top =-1;
    public void push(int n){
       if(top==arr.length-1){
        System.out.println("the stack is full");
       }
       else{
        top++;
        arr[top]=n;
       }

    }

    public void pop(){
        if(top==-1){
            System.out.println("the stack is empty not possible to pop");
        }
        else{
            System.out.println("the popped element is"+arr[top]);
            top--;
        }
    }

    public void display(){
        if(top==-1){
            System.out.println("the stack is empty nothing to dispaly");
        }
        else{
            for(int i=0;i<=top;i++){
                System.out.println(arr[i]);
            }
        }
    }

    public static void main(String []args){
        stack o = new stack();
        o.push(2);
        o.push(24);
        o.push(27);
        o.push(23);
        o.push(32);
        o.push(29);
        o.pop();
        o.pop();
        o.display();
    }
}

File: data structures/stack/stack.py
class stack :
    
    def __init__(self,n) :
        self.list = [0]*n
        self.i = 0
        self.size = n
    def push(self,a) :
        if(self.i!=self.size):
            self.list[self.i]=a
            self.i+=1
        else :
            print("the stack is full")

    def pop(self) :
        if self.i==0 :
            print("the stack is empty nothing to print")
        else :
            self.list.pop()
            self.i-=1
    def display(self):
        if self.i==0 :
            print("the stack is empty")
        else :
            for j in range(0,self.i):
               print(self.list[j])
    
def main() :
   st = stack(5)
   st.push(22)
   st.push(433)
   st.push(33)
   st.display()
   print("after popping")
   st.pop()
   st.display()
main()